model/project_event.go

```diff
@@ -108,6 +112,24 @@ func getRedactedVarsCopy(vars map[string]string, modifiedVarNames map[string]str
 	return redactedVars
 }
 
+// getRedactedGitHubAppCopy returns a copy of the GitHub app auth with the
+// GitHub app's private key redacted if it is set.
+// This intentionally makes a copy to avoid potentially modifying the original
+// GitHub app auth, which may be shared by the actual project settings. That
+// way, callers can still access the unredacted auth credentials.
+func getRedactedGitHubAppCopy(auth githubapp.GithubAppAuth, isGHAppKeyModified bool, placeholder string) githubapp.GithubAppAuth {
+	if len(auth.PrivateKey) == 0 {
+		return auth
+	}
+	redactedAuth := auth
+	if len(redactedAuth.PrivateKey) > 0 && isGHAppKeyModified {
```

Comment:
we don't really need to check` len(redactedAuth.PrivateKey) > 0` here because we already know that `len(auth.PrivateKey) == 0` is not true.

Reply:
We do need to check if the value is empty because if the app was just created, the UI should display the before value as "" and the after value as `{REDACTED_AFTER}` (and vice versa for deletion). Since the redaction logic is shared by both the before and after copy of the project settings, we need to distinguish if it was initially empty or changed to empty.